fe/src/test/java/org/apache/doris/planner/QueryPlanTest.java

```diff
@@ -504,4 +504,106 @@ public void testDateTypeEquality() throws Exception {
         Catalog.getCurrentCatalog().getLoadManager().createLoadJobV1FromStmt(loadStmt, EtlJobType.HADOOP,
                 System.currentTimeMillis());
     }
+
+    private SelectStmt getAnalyzedAndRewritedStmt(String sql) throws Exception {
```

Comment:
Don't need to add this method. you could use `getSQLPlanOrErrorMsg` method.

Reply:
```getSQLPlanOrErrorMsg ```  is unable to meet demand.What we want is to check whether the plan changes as our wish,so we need a  tools not only retwrite the plan but alse contains the customized check logic.It's obviously that plain sql is directly,but not a common general method,because the print format may change.
Maybe list a todo here?
Anyway,I think the result of ```getSQLPlanOrErrorMsg ```  can't meet demand

Reply:
I don't think so.
Directly check final FE sql plan is more accurate:
1. FE send BE is the sql plan
2. Your work do indeed is changing partial plan to constant.

> What we want is to check whether the plan changes as our wish

So you should use `getSQLPlanOrErrorMsg `

> I think the result of getSQLPlanOrErrorMsg can't meet demand

Why couldn't meet you demand? I think which could.